From 31437c96886d64fffc41ecd3db2922e301380c0d Mon Sep 17 00:00:00 2001
From: John Youn <johnyoun@synopsys.com>
Date: Mon, 23 May 2016 11:32:45 -0700
Subject: [PATCH 10/13] usb: dwc3: gadget: Fix truncated cast issue

From sparse:
warning: cast truncates bits from constant value (100 becomes 0)

The DWC3_TRB_NUM constant is too big for u8. Do the calculation a
slightly different way that should still be optimized out for the case
where DWC3_TRB_NUM == 256.

Signed-off-by: John Youn <johnyoun@synopsys.com>
Signed-off-by: Felipe Balbi <felipe.balbi@linux.intel.com>
(cherry picked from commit 02c6ef28866bb7c4bc278180b9c0d2d4e90030bb)
Signed-off-by: Janusz Dziedzic <januszx.dziedzic@intel.com>

Change-Id: I4c7d7867dd2e541cff3d245ffa37711b394b8df3
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-34491
Reviewed-on: https://android.intel.com:443/519134
---
 drivers/usb/dwc3/gadget.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index f51e250..75f23ea 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -916,7 +916,7 @@ static u32 dwc3_calc_trbs_left(struct dwc3_ep *dep)
 	}
 
 	trbs_left = dep->trb_dequeue - dep->trb_enqueue;
-	trbs_left %= DWC3_TRB_NUM;
+	trbs_left &= (DWC3_TRB_NUM - 1);
 
 	if (dep->trb_dequeue < dep->trb_enqueue)
 		trbs_left--;
-- 
1.9.1

